src/Controller/FigureFormController.php

```diff
@@ -11,20 +17,48 @@ class FigureFormController extends AbstractController
     /**
      * @Route("/figure/form", name="create_figure")
      */
-    public function createFigure(): Response
+    public function createFigure(Request $request, EntityManagerInterface $entityManager, ImageFormHandler $imageFormHandler, VideoFormHandler $videoFormHandler): Response
     {
+        $figure = new Figure();
+        $form = $this->createForm(FigureFormType::class, $figure);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $imageFormHandler->handleImageForm($form, $figure);
+            $videoFormHandler->handleVideoForm($form, $figure);
+
+            $entityManager->persist($figure);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('homepage');
+        }
+
         return $this->render('figure_form.html.twig', [
             'controller_name' => 'FigureFormController',
+            'figure_form' => $form->createView(),
         ]);
     }
 
     /**
      * @Route("/figure/form/{slug}", name="modify_figure")
      */
-    public function modifyFigure(): Response
+    public function modifyFigure(Request $request, Figure $figure, EntityManagerInterface $entityManager, ImageFormHandler $imageFormHandler, VideoFormHandler $videoFormHandler): Response
     {
+        $form = $this->createForm(FigureFormType::class, $figure);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $imageFormHandler->handleImageForm($form, $figure);
+            $videoFormHandler->handleVideoForm($form, $figure);
+
+            $entityManager->persist($figure);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('homepage');
+        }
+
         return $this->render('figure_form.html.twig', [
             'controller_name' => 'FigureFormController',
+            'figure_form' => $form->createView(),
+            'figure' => $figure,
         ]);
     }
```

Comment:
These two actions could be merged into one. Or at least, in a simpler way, the duplicated code could be externalized in a private method or better yet a service.

Reply:
Ok, got, it. I would do it on the creation and modification form branch. I should do a stash in the wrong way... This methods shouldn't be on these branch, sorry...